attest/application_key.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2021 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package attest
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rsa"
+	"fmt"
+	"io"
+)
+
+type appKey interface {
+	close(tpmBase) error
+	marshal() ([]byte, error)
+	certificationParameters() CertificationParameters
+	sign(tpmBase, []byte) ([]byte, error)
+	decrypt(tpmBase, []byte) ([]byte, error)
+}
+
+// ApplicationKey represents a key which can be used for signing and decrypting
+// outside-TPM objects.
+type ApplicationKey struct {
```

Comment:
nit: not a huge deal, but I noticed that ApplicationKey seems like a long name. Same with NewAppKey, LoadAppKey, etc. What if we just named this "Key"? So it'd be

- NewKey()
- LoadKey()
- KeyConfig
- etc.

Does that seem easier? Also fine with just sticking with ApplicationKey.

Reply:
`Key` sounds much better, renamed ;-).
All other comments are addressed too.